src/Unitverse.Core.Tests/Helpers/InvocationExtractorTests.cs

```diff
@@ -25,6 +26,17 @@ public void CanCallExtractFrom()
             result.GetAccessedMethodSymbolsFor("_dummyService2").Select(x => x.Name).Should().BeEquivalentTo("ReturnMethod");
         }
 
+        [Test]
+        public void ExtractFrom_DependencyCalledInsidePrivateMethod_ReturnsCalledMethods()
+        {
+            var classModel = ClassModelProvider.CreateModel(TestClasses.AutomaticMockGeneration);
```

Comment:
This gets it's source from https://github.com/mattwhitfield/Unitverse/blob/master/src/Unitverse.Core.Tests/Resources/AutomaticMockGeneration.txt - I made an earlier comment about the example generation (that will serve me right for trying to review on my phone!) - but I think it would be worth editing AutomaticMockGeneration.txt in the Unitverse.Core.Tests project to add a few more sample use cases in there.

Reply:
Also, I think the code as written will pick up dependencies called from within a delegate, so it would be good to try and add some asserts here that check that

Reply:
I added those tests. It really does cover those cases! Should I also update ExampleGenerator with those cases?

Reply:
That's up to you, but I would do so to show it off :)